projectq/backends/_awsbraket/_awsbraket.py

```diff
@@ -0,0 +1,481 @@
+        """
+        # finally: measurements (no intermediate measurements are allowed)
+        # No explicit measurements have to be writen down. To run the circuit
+        # make implicit measurement of all the qubits
+
+        # In Braket the results for the jobs are stored in S3.
+        # You can recover the results from previous jobs using the
+        # TaskArn (self._retrieve_execution).
+        # Keept the code for retreive here to maintain the same strcuture
+        try:
+            if self._retrieve_execution is not None:
+                res = retrieve(credentials=self._credentials,
+                               taskArn=self._retrieve_execution,
+                               num_retries=self._num_retries,
+                               interval=self._interval,
+                               verbose=self._verbose)
+            else:
+                # Return if no operations are added.
+                if self._circuit == "":
+                    return
+
+                n_qubit = max(self._allocated_qubits) + 1
+                info = {}
+                info['circuit'] = self._circuithead + \
+                    self._circuit.rstrip(', ') + \
+                    self._circuittail
+                info['nq'] = n_qubit
+                info['shots'] = self._num_runs
+                info['backend'] = {'name': self.device}
+                res = send(info,
+                           device=self.device,
+                           credentials=self._credentials,
+                           s3_folder=self._s3_folder,
+                           num_retries=self._num_retries,
+                           interval=self._interval,
+                           verbose=self._verbose)
+
+            counts = res
+
+            # Determine random outcome
+            P = random.random()
+            p_sum = 0.
+            measured = ""
+            length = len(self._measured_ids)
+            for state in counts:
+                probability = counts[state]
+                p_sum += probability
+                star = ""
+                if p_sum >= P and measured == "":
+                    measured = state
+                    star = "*"
+                self._probabilities[state] = probability
+                if self._verbose and probability > 0:
+                    print(state + " with p = " + str(probability) +
+                          star)
+
+            class QB():
+                def __init__(self, qubit_id):
+                    self.id = qubit_id
+
+            # register measurement result
+            for qubit_id in self._measured_ids:
+                location = self._logical_to_physical(qubit_id)
+                result = int(measured[location])
+                self.main_engine.set_measurement_result(QB(qubit_id), result)
+            self._reset()
+        except TypeError:
+            raise Exception("Failed to run the circuit. Aborting.")
+
+    def receive(self, command_list):
```

Comment:
I did some tests and this is already working (without adding anything). for example:

```
backend = _AWSBraketBackend(credentials=creds, blah, blah...)
eng = MainEngine(backend=backend, engine_list=[CommandPrinter()])

Some qubits
Some circuit
eng.flush()
```

And In addition to the call to AWS braket I get the commands printed as expected from the CommandPrinter. I used DummyEngine (nothing happened as expected) and ResourceCounter (get the gate an qubit counts) as well without error.

I understand that this is because BasicEngine is already managing that and we have AWSBraketBackend(BasicEngine) (same for IBMBackend and AQTBackend)

Is that what you were asking for? Please let me know

Reply:
I was thinking of that case:

```python
awsbackend = _AWSBraketBackend(credentials=creds, blah, blah...)
eng = MainEngine(backend=CommandPrinter(), engine_list=[awsbackend])


# ... some operations ...
eng.flush()
```

Reply:
Just added the same that in the Simulator backend to make it work in the case you mention.

I will push the updates during the weekend

Thanks